src/components/templates/PartyMap/PartyMap.tsx

```diff
@@ -12,74 +12,183 @@ import { useSelector } from "hooks/useSelector";
 
 import { Map, RoomModal } from "./components";
 
-import "./PartyMap.scss";
 import AnnouncementMessage from "components/molecules/AnnouncementMessage/AnnouncementMessage";
 import SparkleFairiesPopUp from "components/molecules/SparkleFairiesPopUp/SparkleFairiesPopUp";
 
+import "./PartyMap.scss";
+
 const partyMapVenueSelector = (state: RootState) =>
   state.firestore.ordered.currentVenue?.[0] as PartyMapVenue;
 
 export const PartyMap: React.FC = () => {
-  const [isRoomModalOpen, setIsRoomModalOpen] = useState(false);
+  const currentVenue = useSelector(partyMapVenueSelector);
+
+  // TODO: can we just make this a boolean that is set from isTruthy(selectedRoom)?
+  const [isRoomModalOpen, setRoomModalOpen] = useState(false);
   const [selectedRoom, setSelectedRoom] = useState<
     PartyMapRoomData | undefined
   >();
 
-  const venue = useSelector(partyMapVenueSelector);
-  const usersInCamp = useCampPartygoers(venue.name);
-
-  const attendances = useMemo(
-    () =>
-      usersInCamp
-        ? usersInCamp.reduce<Record<string, number>>((acc, value) => {
-            Object.keys(value.lastSeenIn).forEach((key) => {
-              acc[key] = (acc[key] || 0) + 1;
-            });
-            return acc;
-          }, {})
-        : {},
-    [usersInCamp]
-  );
+  const selectRoom = useCallback((room: PartyMapRoomData) => {
+    setSelectedRoom(room);
+    setRoomModalOpen(true);
+  }, []);
 
-  const modalHidden = useCallback(() => {
-    setIsRoomModalOpen(false);
+  const unselectRoom = useCallback(() => {
+    // TODO: should we rework this so we only unselect if it matches? To prevent race conditions?
+    // setRoomClicked((prevRoomClicked) =>
+    //   prevRoomClicked === room.title ? undefined : room.title
+    // );
+    setSelectedRoom(undefined);
+    setRoomModalOpen(false);
   }, []);
 
+  // TODO: can we get rid of this in favour of unselectRoom?
+  const closeRoomModal = useCallback(() => {
```

Comment:
I'm not sure, it depends on the desirect UX, there were a lot of changes and at some point we wanted to have a selected room when we are underneath it but not show the modal, i'm not exactly 100% sure what the final goal/product is but we might need the 2 states separately.